### PR TITLE
fix for TIKA-2021 contributed by Zarana Parekh

### PR DESCRIPTION
**Pull request by Zarana-Parekh**
Improving accuracy of Tesseract for better extraction of numeric and alphanumeric text from images.

### CODE REVIEW

File: tika-bundle/pom.xml [hunk 1]
```diff
@@ -426,6 +427,7 @@
         </configuration>
       </plugin>
     </plugins>
+    </pluginManagement>
```
Comment (chrismattmann) [inline tika-bundle/pom.xml]: please remove this change, it's orthogonal and just whitespace.

File: tika-bundle/pom.xml [hunk 2]
```diff
@@ -112,6 +112,7 @@
   </dependencies>
 
   <build>
+  	<pluginManagement>
```
Comment (chrismattmann) [inline tika-bundle/pom.xml]: please remove this change, it's orthogonal and just whitespace.

File: tika-parsers/pom.xml [hunk 3]
```diff
@@ -502,6 +509,7 @@
         </file>
       </activation>
       <build>
+      <pluginManagement>
```
Comment (chrismattmann) [inline tika-parsers/pom.xml]: orthogonal remove

File: tika-parsers/pom.xml [hunk 4]
```diff
@@ -532,6 +540,7 @@
             </executions>
           </plugin>
         </plugins>
+        </pluginManagement>
```
Comment (chrismattmann) [inline tika-parsers/pom.xml]: orthogonal remove

File: tika-parsers/src/main/java/org/apache/tika/parser/ocr/TesseractOCRConfig.java [hunk 5]
```diff
@@ -113,9 +132,23 @@ private void init(InputStream is) {
...
+		setColorspace(
+				getProp(props, "colorspace", getColorspace()));
+		setFilter(
+				getProp(props, "filter", getFilter()));
+		setResize(
+				getProp(props, "resize", getResize()));
 
 	}
-
+
```
Comment (chrismattmann) [inline tika-parsers/src/main/java/org/apache/tika/parser/ocr/TesseractOCRConfig.java]: orthogonal remove

**Comment by chrismattmann** [comment]
@Zarana-Parekh looking great! one question (after you address the easy ones above) - will this be enabled by default? The reason I ask is that I think this will introduce quite a bit of overhead into processing and probably shouldn't be on by default.

**Comment by Zarana-Parekh** [comment]
@chrismattmann Thanks for the feedback. I will incorporate the above changes. The current configuration has it enabled by default but will modify that to make it optional.

File: tika-bundle/pom.xml [hunk 6]
```diff
@@ -112,320 +112,320 @@
   </dependencies>
 
   <build>
-    <plugins>
```
Comment (chrismattmann) [inline tika-bundle/pom.xml]: orthogonal change - can you remove these orthogonal changes?

File: tika-parsers/src/main/resources/org/apache/tika/parser/ocr/TesseractOCRConfig.properties [hunk 7]
```diff
@@ -13,9 +13,20 @@
...
 pageSegMode=1
 maxFileSizeToOcr=2147483647
 minFileSizeToOcr=0
-timeout=120
\ No newline at end of file
+timeout=120
+
+# properties for image processing
+# to enable processing, set enableProcessing to 1
+enableProcessing=0
```
Comment (chrismattmann) [inline tika-parsers/src/main/resources/org/apache/tika/parser/ocr/TesseractOCRConfig.properties]: can we think of something besides "processing?" that's too generic. how about `enableImageProcessing`

File: tika-parsers/src/test/resources/test-properties/TesseractOCRConfig-partial.properties [hunk 8]
```diff
@@ -15,4 +15,10 @@
 
 language=fra+deu
 timeout=240
-minFileSizeToOcr=1
\ No newline at end of file
+minFileSizeToOcr=1
+
+enableProcessing=0
```
Comment (chrismattmann) [inline tika-parsers/src/test/resources/test-properties/TesseractOCRConfig-partial.properties]: also how about `enableImageProcessing`?

**Comment by chrismattmann** [comment]
a lot better @Zarana-Parekh ! now just need to remove an orthogonal change (again) on the tika-bundle pom.xml, and also consider changing the property name.

File: tika-bundle/pom.xml (hunk 6 above)
Comment (lewismc) [inline tika-bundle/pom.xml]: ACK @Zarana-Parekh please rebase tika-bundle/pom.xml against master branch as the formatting changes make the patch contribution unclear.

File: tika-parsers/pom.xml [hunk 9]
```diff
@@ -256,7 +257,13 @@
       <artifactId>cxf-rt-rs-client</artifactId>
       <version>${cxf.version}</version>
     </dependency>
-	
+    <!-- TIKA-2021: Tesseract OCR Parser dependencies, 
+    used for executing image processing script -->
+	<dependency>
+	    <groupId>org.apache.commons</groupId>
+	    <artifactId>commons-exec</artifactId>
+	    <version>${commonsexec.version}</version>
```
Comment (lewismc) [inline tika-parsers/pom.xml]: Can you add scope to compile

File: tika-parsers/src/main/java/org/apache/tika/parser/ocr/TesseractOCRConfig.java [hunk 10]
```diff
@@ -222,8 +260,143 @@ public void setTimeout(int timeout) {
 	public int getTimeout() {
 		return timeout;
 	}
+	
+	/** @see #setEnableImageProcessing(boolean) */
```
Comment (lewismc) [inline tika-parsers/src/main/java/org/apache/tika/parser/ocr/TesseractOCRConfig.java]: Please include @return Javadoc

File: tika-parsers/src/main/java/org/apache/tika/parser/ocr/TesseractOCRConfig.java [hunk 11]
```diff
@@ -222,8 +260,143 @@ public void setTimeout(int timeout) {
...
+	 * Default value is false.
+	 */
+	public void setEnableImageProcessing(int enableImageProcessing) {
+		this.enableImageProcessing = enableImageProcessing;
+	}
+
+	/**
+	 * @return the density
+	 */
+	public int getDensity() {
```
Comment (lewismc) [inline tika-parsers/src/main/java/org/apache/tika/parser/ocr/TesseractOCRConfig.java]: Please add @Return javadoc

File: tika-parsers/src/main/java/org/apache/tika/parser/ocr/TesseractOCRConfig.java [hunk 12]
```diff
@@ -222,8 +260,143 @@ public void setTimeout(int timeout) {
...
+		return density;
+	}
+
+	/**
+	 * @param density the density to set
```
Comment (lewismc) [inline tika-parsers/src/main/java/org/apache/tika/parser/ocr/TesseractOCRConfig.java]: Please add the range that a density integer value can be

File: tika-parsers/src/main/java/org/apache/tika/parser/ocr/TesseractOCRConfig.java [hunk 13]
```diff
@@ -222,8 +260,143 @@ public void setTimeout(int timeout) {
...
+	 * Default value is 300.
+	 */
+	public void setDensity(int density) {
+		if(density < 150 || density > 1200) {
+			throw new IllegalArgumentException("Invalid density value");
```
Comment (lewismc) [inline tika-parsers/src/main/java/org/apache/tika/parser/ocr/TesseractOCRConfig.java]: Please indicate possible values in Exception message

File: tika-parsers/src/main/java/org/apache/tika/parser/ocr/TesseractOCRConfig.java [hunk 14]
```diff
@@ -222,8 +260,143 @@ public void setTimeout(int timeout) {
...
+	 */
+	public void setDepth(int depth) {
+		int[] allowedValues = {2, 4, 8, 16, 32, 64, 256, 4096};
+		for (int i = 0; i < allowedValues.length; i++) {
+			if(depth == allowedValues[i]) {
+				this.depth = depth;
+				return;
+			}
+		}
+		throw new IllegalArgumentException("Invalid depth value");
```
Comment (lewismc) [inline tika-parsers/src/main/java/org/apache/tika/parser/ocr/TesseractOCRConfig.java]: Please indicate possible values in Exception message

File: tika-parsers/src/main/java/org/apache/tika/parser/ocr/TesseractOCRConfig.java [hunk 15]
```diff
@@ -222,8 +260,143 @@ public void setTimeout(int timeout) {
...
 
 	/**
+	 * @return the depth
+	 */
+	public int getDepth() {
+		return depth;
+	}
+
+	/**
+	 * @param depth the depth to set
```
Comment (lewismc) [inline tika-parsers/src/main/java/org/apache/tika/parser/ocr/TesseractOCRConfig.java]: Please indicate possible values in Javadoc

File: tika-parsers/src/main/java/org/apache/tika/parser/ocr/TesseractOCRConfig.java [hunk 16]
```diff
@@ -222,8 +260,143 @@ public void setTimeout(int timeout) {
...
+
+	/**
+	 * @param colorspace the colorspace to set
+	 * Deafult value is gray.
+	 */
+	public void setColorspace(String colorspace) {
+		if(!colorspace.equals(null)) {
+			this.colorspace = colorspace;
+		} else {
+			throw new IllegalArgumentException("Invalid colorspace value");
```
Comment (lewismc) [inline tika-parsers/src/main/java/org/apache/tika/parser/ocr/TesseractOCRConfig.java]: This Exception message is misleading. The value is not invalid... it is simply null. The message should state "Colorspace value cannot be null"

File: tika-parsers/src/main/java/org/apache/tika/parser/ocr/TesseractOCRConfig.java [hunk 17]
```diff
@@ -222,8 +260,143 @@ public void setTimeout(int timeout) {
...
+		return filter;
+	}
+
+	/**
+	 * @param filter the filter to set
+	 * Default value is triangle.
+	 */
+	public void setFilter(String filter) {
+		if(filter.equals(null)) {
+			throw new IllegalArgumentException("Invalid filter value");
```
Comment (lewismc) [inline tika-parsers/src/main/java/org/apache/tika/parser/ocr/TesseractOCRConfig.java]: This Exception message is misleading. The value is not invalid... it is simply null. The message should state "Filter value cannot be null"

File: tika-parsers/src/main/java/org/apache/tika/parser/ocr/TesseractOCRConfig.java [hunk 18]
```diff
@@ -222,8 +260,143 @@ public void setTimeout(int timeout) {
...
+		}
+		
+		String[] allowedFilters = {"Point", "Hermite", "Cubic", "Box", "Gaussian", "Catrom", "Triangle", "Quadratic", "Mitchell"};
+		for (int i = 0; i < allowedFilters.length; i++) {
+			if(filter.equalsIgnoreCase(allowedFilters[i])) {
+				this.filter = filter;
+				return;
+			}
+		}
+		throw new IllegalArgumentException("Invalid filter value");
```
Comment (lewismc) [inline tika-parsers/src/main/java/org/apache/tika/parser/ocr/TesseractOCRConfig.java]: You may also wish to print the allowedFilters as part of the exception message.

File: tika-parsers/src/main/java/org/apache/tika/parser/ocr/TesseractOCRConfig.java [hunk 19]
```diff
@@ -222,8 +260,143 @@ public void setTimeout(int timeout) {
...
+	 * Default value is 900.
+	 */
+	public void setResize(int resize) {
+		for(int i=1;i<10;i++) {
+			if(resize == i*100) {
+				this.resize = resize;
+				return;
+			}
+		}
+		throw new IllegalArgumentException("Invalid resize value");
```
Comment (lewismc) [inline tika-parsers/src/main/java/org/apache/tika/parser/ocr/TesseractOCRConfig.java]: You may also wish to print the valid range of resize values as part of the exception message.

File: tika-parsers/src/main/java/org/apache/tika/parser/ocr/TesseractOCRConfig.java [hunk 20]
```diff
@@ -222,8 +260,143 @@ public void setTimeout(int timeout) {
...
+	/** @see #setImageMagickPath(String ImageMagickPath)*/
+	public String getImageMagickPath() {
+		
+		return ImageMagickPath;
+	}
+	
+	/**
+	 * Set the path to the ImageMagick executable, needed if it is not on system path.
+	 */
+	public void setImageMagickPath(String ImageMagickPath) {
```
Comment (lewismc) [inline tika-parsers/src/main/java/org/apache/tika/parser/ocr/TesseractOCRConfig.java]: Please add @param for Javadoc

File: tika-parsers/src/main/java/org/apache/tika/parser/ocr/TesseractOCRParser.java [hunk 21]
```diff
@@ -127,7 +136,43 @@ public boolean hasTesseract(TesseractOCRConfig config) {
         return hasTesseract;
      
     }
+    
+    public boolean hasImageMagick(TesseractOCRConfig config) {
```
Comment (lewismc) [inline tika-parsers/src/main/java/org/apache/tika/parser/ocr/TesseractOCRParser.java]: Please add Javadoc to all public methods

File: tika-parsers/src/main/java/org/apache/tika/parser/ocr/TesseractOCRParser.java [hunk 22]
```diff
@@ -127,7 +136,43 @@ public boolean hasTesseract(TesseractOCRConfig config) {
...
+        // Try running ImageMagick program from there, and see if it exists + works
+        String[] checkCmd = { ImageMagick };
+        boolean hasImageMagick = ExternalParser.check(checkCmd);
+        TESSERACT_PRESENT.put(ImageMagick, hasImageMagick);
+        
+        return hasImageMagick;
+     
+    }
+    
+    public boolean hasPython() {
```
Comment (lewismc) [inline tika-parsers/src/main/java/org/apache/tika/parser/ocr/TesseractOCRParser.java]: Please add Javadoc to all public methods

File: tika-parsers/src/main/java/org/apache/tika/parser/ocr/TesseractOCRParser.java [hunk 23]
```diff
@@ -212,6 +257,52 @@ public void parseInline(InputStream stream, XHTMLContentHandler xhtml, Tesseract
 
     }
 
+    /**
+     * This method is used to process the image to an OCR-friendly format.
+     * @param streamingObject input image to be processed
+     * @param config TesseractOCRconfig class to get ImageMagick properties
+     * @throws IOException
```
Comment (lewismc) [inline tika-parsers/src/main/java/org/apache/tika/parser/ocr/TesseractOCRParser.java]: Why would it throw IOException? Please substantiate on Javadoc

File: tika-parsers/src/main/java/org/apache/tika/parser/ocr/TesseractOCRParser.java [hunk 24]
```diff
@@ -212,6 +257,52 @@ public void parseInline(InputStream stream, XHTMLContentHandler xhtml, Tesseract
...
+     * @throws TikaException
```
Comment (lewismc) [inline tika-parsers/src/main/java/org/apache/tika/parser/ocr/TesseractOCRParser.java]: Why would it throw TikaException? Please substantiate on Javadoc

File: tika-parsers/src/main/java/org/apache/tika/parser/ocr/TesseractOCRParser.java [hunk 25]
```diff
@@ -212,6 +257,52 @@ public void parseInline(InputStream stream, XHTMLContentHandler xhtml, Tesseract
...
+    			
+    	DefaultExecutor executor = new DefaultExecutor();
+    	ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+    	PumpStreamHandler streamHandler = new PumpStreamHandler(outputStream);
+        executor.setStreamHandler(streamHandler);
+        
+        // determine the angle of rotation required to make the text horizontal
+        CommandLine cmdLine = CommandLine.parse(cmd);
+        if(hasPython()) {
+			try {
```
Comment (lewismc) [inline tika-parsers/src/main/java/org/apache/tika/parser/ocr/TesseractOCRParser.java]: Formatting is not correct here. Please use 4 space indents

File: tika-parsers/src/main/java/org/apache/tika/parser/ocr/TesseractOCRParser.java [hunk 26]
```diff
@@ -212,6 +257,52 @@ public void parseInline(InputStream stream, XHTMLContentHandler xhtml, Tesseract
...
+				executor.execute(cmdLine);
+				angle = outputStream.toString().trim();
+			} catch(Exception e) {	
+				e.printStackTrace();
+			}
+        }
+              
+        // process the image - parameter values can be set in TesseractOCRConfig.properties
+    	String line = "convert -density " + config.getDensity() + " -depth " + config.getDepth() + " -colorspace " + config.getColorspace() +  " -filter " + config.getFilter() + " -resize " + config.getResize() + "% -rotate "+ angle + " " + streamingObject.getAbsolutePath() + " " + streamingObject.getAbsolutePath();
```
Comment (lewismc) [inline tika-parsers/src/main/java/org/apache/tika/parser/ocr/TesseractOCRParser.java]: Split this into separate lines

File: tika-parsers/src/main/resources/org/apache/tika/parser/ocr/rotation.py [hunk 27]
```diff
@@ -0,0 +1,72 @@
...
+from matplotlib.mlab import rms_flat
+
+import sys
+import getopt
+
+def main(argv):
+	filename = ''
+	
+	if len(sys.argv) < 3:
+		print('Usage: rotation_spacing.py -f <filename>')
```
Comment (lewismc) [inline tika-parsers/src/main/resources/org/apache/tika/parser/ocr/rotation.py]: This script is not called rotation_spacing.py but instead rotation.py right?

File: tika-parsers/src/main/resources/org/apache/tika/parser/ocr/rotation.py [hunk 28]
```diff
@@ -0,0 +1,72 @@
...
+		sys.exit()
+	try:
+	  opts, args = getopt.getopt(argv,"hf:",["file="])
+	except getopt.GetoptError:
+	  print('rotation_spacing.py -f <filename>')
```
Comment (lewismc) [inline tika-parsers/src/main/resources/org/apache/tika/parser/ocr/rotation.py]: This script is not called rotation_spacing.py but instead rotation.py right?

File: tika-parsers/src/main/resources/org/apache/tika/parser/ocr/rotation.py [hunk 29]
```diff
@@ -0,0 +1,72 @@
...
+	  sys.exit(2)
+	for opt, arg in opts:
+	  if opt == '-h':
+	     print('Usage: rotation_spacing.py -f <filename>')
```
Comment (lewismc) [inline tika-parsers/src/main/resources/org/apache/tika/parser/ocr/rotation.py]: This script is not called rotation_spacing.py but instead rotation.py right?

File: tika-parsers/src/main/resources/org/apache/tika/parser/ocr/rotation.py [hunk 30]
```diff
@@ -0,0 +1,72 @@
...
+	     sys.exit()
+	  elif opt in ("-f", "--file"):
+	     filename = arg
+
+	try:
+		from parabolic import parabolic
```
Comment (lewismc) [inline tika-parsers/src/main/resources/org/apache/tika/parser/ocr/rotation.py]: Please check formatting. Should be set to 4 space indents

File: tika-parsers/src/main/resources/org/apache/tika/parser/ocr/rotation.py [hunk 31]
```diff
@@ -0,0 +1,72 @@
...
+
+		def argmax(x):
+		    return parabolic(x, numpy.argmax(x))[0]
+	except ImportError:
+		from numpy import argmax
```
Comment (lewismc) [inline tika-parsers/src/main/resources/org/apache/tika/parser/ocr/rotation.py]: Is this import an error?

File: tika-parsers/src/test/java/org/apache/tika/parser/ocr/TesseractOCRConfigTest.java [hunk 32]
```diff
@@ -70,6 +81,11 @@ public void testFullConfig() throws Exception {
         assertEquals("Invalid overridden minFileSizeToOcr value", 1, config.getMinFileSizeToOcr());
         assertEquals("Invalid overridden maxFileSizeToOcr value", 2000000, config.getMaxFileSizeToOcr());
         assertEquals("Invalid overridden timeout value", 240, config.getTimeout());
+        assertEquals("Invalid overridden ImageMagickPath value", "/usr/local/bin/", config.getImageMagickPath());
```
Comment (lewismc) [inline tika-parsers/src/test/java/org/apache/tika/parser/ocr/TesseractOCRConfigTest.java]: This will not pass on a non-unix OS

File: tika-parsers/src/test/resources/test-properties/TesseractOCRConfig-full.properties [hunk 33]
```diff
@@ -20,3 +20,10 @@ pageSegMode=2
 maxFileSizeToOcr=2000000
 timeout=240
 minFileSizeToOcr=1
+
+enableProcessing=1
+ImageMagickPath=/usr/local/bin
```
Comment (lewismc) [inline tika-parsers/src/test/resources/test-properties/TesseractOCRConfig-full.properties]: I think that this should be set to null value. This will not work on Window for example.

File: tika-parsers/pom.xml (hunk 9 above)
Comment (Zarana-Parekh) [inline tika-parsers/pom.xml]: added scope in fa30edd4d95fc0cdc4492c711f402e24dba24e8b

File: tika-parsers/src/test/resources/test-properties/TesseractOCRConfig-partial.properties (hunk 8 above)
Comment (Zarana-Parekh) [inline tika-parsers/src/test/resources/test-properties/TesseractOCRConfig-partial.properties]: resolved in bc6667c195bbef1a672fd635ff7e253143ba2ad8

File: tika-parsers/src/main/java/org/apache/tika/parser/ocr/TesseractOCRConfig.java (hunk 11 above)
Comment (Zarana-Parekh) [inline tika-parsers/src/main/java/org/apache/tika/parser/ocr/TesseractOCRConfig.java]: added @Return in 6773d42de77230dff621a1010ed37f0505dfa302

File: tika-bundle/pom.xml [hunk 34]
```diff
@@ -440,4 +440,4 @@
     <system>Jenkins</system>
     <url>https://builds.apache.org/job/Tika-trunk/</url>
   </ciManagement>
-</project>
+</project>
```
Comment (chrismattmann) [inline tika-bundle/pom.xml]: seems to be an orthogonal change please remove.

File: tika-parsers/src/test/java/org/apache/tika/parser/ocr/TesseractOCRConfigTest.java (hunk 32 above)
Comment (chrismattmann) [inline tika-parsers/src/test/java/org/apache/tika/parser/ocr/TesseractOCRConfigTest.java]: @Zarana-Parekh can you add a check using Apache commons-util to make sure this is a UNIX OS before running the above test.

File: tika-parsers/src/test/resources/test-properties/TesseractOCRConfig-partial.properties (hunk 8 above)
Comment (chrismattmann) [inline tika-parsers/src/test/resources/test-properties/TesseractOCRConfig-partial.properties]: yep resolved in the main properties file, @Zarana-Parekh but not this test one please update this too.

**Comment by chrismattmann** [comment]
few last simple updates, @Zarana-Parekh then this should be good to go.

**Comment by chrismattmann** [comment]
LGTM I will commit this now! :) 👍 @Zarana-Parekh

**Comment by chrismattmann** [comment]
OK I can't commit this - tests aren't passing:

```
Running org.apache.tika.utils.ServiceLoaderUtilsTest
Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.007 sec - in org.apache.tika.utils.ServiceLoaderUtilsTest

Results :

Failed tests: 
  TesseractOCRParserTest.testImageMagick:183 expected:<true> but was:<false>

Tests run: 730, Failures: 1, Errors: 0, Skipped: 12

[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] Apache Tika parent ................................. SUCCESS [  1.348 s]
[INFO] Apache Tika core ................................... SUCCESS [ 12.065 s]
[INFO] Apache Tika parsers ................................ FAILURE [01:34 min]
[INFO] Apache Tika XMP .................................... SKIPPED
[INFO] Apache Tika serialization .......................... SKIPPED
[INFO] Apache Tika batch .................................. SKIPPED
[INFO] Apache Tika language detection ..................... SKIPPED
```

@Zarana-Parekh ?

**Comment by chrismattmann** [comment]
OK @Zarana-Parekh I made some updates and got the build to pass. Will commit this shortly.

**Comment by chrismattmann** [comment]
fixed:
```
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ tika ---
[INFO] Deleting /Users/mattmann/tmp/tika1.13/target
[INFO] 
[INFO] --- maven-remote-resources-plugin:1.5:process (default) @ tika ---
[INFO] 
[INFO] --- maven-site-plugin:3.4:attach-descriptor (attach-descriptor) @ tika ---
[INFO] 
[INFO] --- forbiddenapis:2.0:check (default) @ tika ---
[INFO] Skipping execution for packaging "pom"
[INFO] 
[INFO] --- forbiddenapis:2.0:testCheck (default) @ tika ---
[INFO] Skipping execution for packaging "pom"
[INFO] 
[INFO] --- maven-install-plugin:2.5.2:install (default-install) @ tika ---
[INFO] Installing /Users/mattmann/tmp/tika1.13/pom.xml to /Users/mattmann/.m2/repository/org/apache/tika/tika/1.14-SNAPSHOT/tika-1.14-SNAPSHOT.pom
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] Apache Tika parent ................................. SUCCESS [  0.916 s]
[INFO] Apache Tika core ................................... SUCCESS [ 10.901 s]
[INFO] Apache Tika parsers ................................ SUCCESS [01:29 min]
[INFO] Apache Tika XMP .................................... SUCCESS [  1.656 s]
[INFO] Apache Tika serialization .......................... SUCCESS [  1.481 s]
[INFO] Apache Tika batch .................................. SUCCESS [01:47 min]
[INFO] Apache Tika language detection ..................... SUCCESS [  2.957 s]
[INFO] Apache Tika application ............................ SUCCESS [ 27.762 s]
[INFO] Apache Tika OSGi bundle ............................ SUCCESS [ 16.128 s]
[INFO] Apache Tika translate .............................. SUCCESS [  1.843 s]
[INFO] Apache Tika server ................................. SUCCESS [ 33.559 s]
[INFO] Apache Tika examples ............................... SUCCESS [  5.806 s]
[INFO] Apache Tika Java-7 Components ...................... SUCCESS [  1.485 s]
[INFO] Apache Tika ........................................ SUCCESS [  0.015 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 05:02 min
[INFO] Finished at: 2016-07-06T23:37:20-07:00
[INFO] Final Memory: 146M/1733M
[INFO] ------------------------------------------------------------------------
LMC-053601:tika1.13 mattmann$ 
```
